src/libs/Navigation/AppNavigator/createCustomBottomTabNavigator/TopBar.tsx

```diff
@@ -17,19 +17,21 @@ import * as Session from '@userActions/Session';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
-import type {Policy} from '@src/types/onyx';
+import type {Policy, Session as SessionType} from '@src/types/onyx';
 
 type TopBarOnyxProps = {
     policy: OnyxEntry<Policy>;
+    session: OnyxEntry<SessionType>;
 };
 
 // eslint-disable-next-line react/no-unused-prop-types
 type TopBarProps = {activeWorkspaceID?: string} & TopBarOnyxProps;
 
-function TopBar({policy}: TopBarProps) {
+function TopBar({policy, session}: TopBarProps) {
     const styles = useThemeStyles();
     const theme = useTheme();
     const {translate} = useLocalize();
+    const isAnonymousUser = session?.authTokenType === CONST.AUTH_TOKEN_TYPES.ANONYMOUS;
```

Comment:
This kind of sucks. Now you duplicated this logic that's also in `isAnonymousUser`.
What can we do so that we do not duplicate this logic and instead reuse it?

Reply:
Do you mean to add the optional `session` here?
https://github.com/Expensify/App/blob/202c0fde4abbfc218bd29e729d1a34b089234d53/src/libs/actions/Session/index.ts#L178-L180

Reply:
Updated